docs/spec/concepts.rst

```diff
@@ -3,97 +3,174 @@
+represented by ``A``. Because the subset relation on sets is transitive and
+reflexive, the subtype relation is also transitive (if ``A`` is a subtype of
+``B`` and ``B`` is a subtype of ``C``, then ``A`` is a subtype of ``C``) and
+reflexive (``A`` is always a subtype of ``A``).
+
+The **supertype** relation is the inverse of subtype: ``A`` is a supertype of
+``B`` if (and only if) ``B`` is a subtype of ``A``; or equivalently, if (and
+only if) the set of values represented by ``A`` is a superset of the values
+represented by ``B``. The supertype relation is also transitive and reflexive.
+
+We also define an **equivalence** relation on static types: the types ``A`` and
+``B`` are equivalent (or "the same type") if (and only if) ``A`` is a subtype
+of ``B`` and ``B`` is a subtype of ``A``. This means that the set of values
+represented by ``A`` is both a superset and a subset of the values represented
+by ``B``, meaning ``A`` and ``B`` must represent the same set of values.
+
+We may describe a type ``B`` as "narrower" than a type ``A`` if ``B`` is a
+subtype of ``A`` and ``B`` is not equivalent to ``A``.
+
+The consistency relation
+------------------------
+
+Since :ref:`Any` (the dynamic type) represents an unknown static type, it does
+not represent any known single set of values, and thus it is not in the domain
+of the subtype, supertype, or equivalence relations on static types described
+above.
+
+We define a **materialization** relation from gradual types to gradual and
+static types as follows: if replacing one or more occurrences of ``Any`` in
+gradual type ``A`` with some gradual or static type results in the gradual or
+static type ``B``, then ``B`` is a materialization of ``A``. For instance,
+``tuple[int, str]`` (a static type) and ``tuple[Any, str]`` (a gradual type)
+are both materializations of ``tuple[Any, Any]``. If ``B`` is a materialization
+of ``A``, we can say that ``B`` is a more precise type than ``A``.
+
+We define a **consistency** relation over all types (the dynamic type, gradual
+types, and static types.)
+
+A static type ``A`` is consistent with another static type ``B`` only if they
+are the same type (``A`` is equivalent to ``B``.)
+
+The dynamic type ``Any`` is consistent with every type, and every type is
+consistent with ``Any``.
+
+A gradual type ``C`` is consistent with a gradual or static type ``D``, and
+``D`` is consistent with ``C``, if ``D`` is a materialization of ``C``.
+
+The consistency relation is not transitive. ``tuple[int, int]`` and
+``tuple[str, int]`` are both consistent with ``tuple[Any, int]]``, but
+``tuple[str, int]`` is not consistent with ``tuple[int, int]``.
+
+The consistency relation is symmetric. If ``A`` is consistent with ``B``, ``B``
+is also consistent with ``A``. It is also reflexive: ``A`` is always consistent
+with ``A``.
+
+The consistent subtype relation
+-------------------------------
+
+Given the consistency relation and the subtyping relation, we define the
```

Comment:
This doesn't use consistency, it uses materialization so you could say "Given the materialization relation and the subtyping relation".

Do we actually need consistency for anything?  We might be able to get away with leaving it out.

Reply:
Good point.

I'd rather not remove consistency just yet, until I've had a chance to go over the rest of the spec and normalize the terms used, so I'm more confident we don't need it. I suspect there may be some cases (e.g. related to invariance, or typevar constraints) where we do need it? But I'm happy to remove it if it turns out we don't need it.

Reply:
Oh yeah, TypeVar constraints was the one place that I thought we needed it.  I also see that I didn't really get consistent subtyping right for invariant or contravariant generics in my doc.